File: backend/platform-core/src/main/java/com/commonsengine/platform/ai/AiServiceProperties.java
package com.commonsengine.platform.ai;

import org.springframework.boot.context.properties.ConfigurationProperties;

import java.time.Duration;

/**
 * Configuration for the AI services integration.
 *
 * <p>Bound from the {@code ai-services.*} prefix in application properties.
 */
@ConfigurationProperties(prefix = "ai-services")
public class AiServiceProperties {

    /** Base URL of the AI service backend. */
    private String baseUrl = "http://localhost:8000";

    /** API key for authentication. */
    private String apiKey = "";

    /** HTTP request timeout. */
    private Duration timeout = Duration.ofSeconds(10);

    /** Circuit-breaker config. */
    private CircuitBreaker circuitBreaker = new CircuitBreaker();

    /** Retry config. */
    private Retry retry = new Retry();

    // ---- nested config ------------------------------------------------

    public static class CircuitBreaker {
        private double failureRateThreshold = 50.0;
        private int slidingWindowSize = 10;
        private int minimumNumberOfCalls = 5;
        private Duration waitDurationInOpenState = Duration.ofSeconds(30);

        public double getFailureRateThreshold() { return failureRateThreshold; }
        public void setFailureRateThreshold(double v) { this.failureRateThreshold = v; }

        public int getSlidingWindowSize() { return slidingWindowSize; }
        public void setSlidingWindowSize(int v) { this.slidingWindowSize = v; }

        public int getMinimumNumberOfCalls() { return minimumNumberOfCalls; }
        public void setMinimumNumberOfCalls(int v) { this.minimumNumberOfCalls = v; }

        public Duration getWaitDurationInOpenState() { return waitDurationInOpenState; }
        public void setWaitDurationInOpenState(Duration v) { this.waitDurationInOpenState = v; }
    }

    public static class Retry {
        private int maxAttempts = 3;
        private Duration waitDuration = Duration.ofMillis(500);

        public int getMaxAttempts() { return maxAttempts; }
        public void setMaxAttempts(int v) { this.maxAttempts = v; }

        public Duration getWaitDuration() { return waitDuration; }
        public void setWaitDuration(Duration v) { this.waitDuration = v; }
    }

    // ---- getters / setters --------------------------------------------

    public String getBaseUrl() { return baseUrl; }
    public void setBaseUrl(String baseUrl) { this.baseUrl = baseUrl; }

    public String getApiKey() { return apiKey; }
    public void setApiKey(String apiKey) { this.apiKey = apiKey; }

    public Duration getTimeout() { return timeout; }
    public void setTimeout(Duration timeout) { this.timeout = timeout; }

    public CircuitBreaker getCircuitBreaker() { return circuitBreaker; }
    public void setCircuitBreaker(CircuitBreaker circuitBreaker) { this.circuitBreaker = circuitBreaker; }

    public Retry getRetry() { return retry; }
    public void setRetry(Retry retry) { this.retry = retry; }
}

File: backend/platform-core/src/main/kotlin/com/commonsengine/platform/ai/AiServiceProperties.kt
package com.commonsengine.platform.ai

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * AI 服务层连接配置。
 *
 * 三个 Python 微服务的 base URL（默认本地开发地址）。生产环境通过
 * 环境变量覆盖（如 `AI_SERVICES_CUSTOMER_SERVICE_BASE_URL`）。
 *
 * 熔断与重试参数遵循 Resilience4j 语义，默认值面向「AI 服务非关键路径」
 * 的保守策略：快速失败、有限重试、降级而非抛出。
 *
 * — Commons Engine Chief Engineer Bot（AI），#74
 */
@ConfigurationProperties(prefix = "ai-services")
data class AiServiceProperties(

    /** 各服务 base URL（无尾斜杠）。 */
    val customerService: ServiceEndpoint = ServiceEndpoint(baseUrl = "http://localhost:8001"),
    val contentModeration: ServiceEndpoint = ServiceEndpoint(baseUrl = "http://localhost:8002"),
    val dispatchOptimizer: ServiceEndpoint = ServiceEndpoint(baseUrl = "http://localhost:8003"),

    /** 连接超时（毫秒）。 */
    val connectTimeoutMs: Int = 1_000,

    /** 读取超时（毫秒）— AI 推理可能较慢，略宽于普通 HTTP。 */
    val readTimeoutMs: Int = 3_000,

    val circuitBreaker: CircuitBreakerConfig = CircuitBreakerConfig(),
    val retry: RetryConfig = RetryConfig(),
) {
    data class ServiceEndpoint(val baseUrl: String)

    data class CircuitBreakerConfig(
        /** 失败率阈值百分比（0-100），达到后打开熔断器。 */
        val failureRateThreshold: Float = 50f,
        /** 熔断器打开后等待多久进入半开状态（秒）。 */
        val waitDurationInOpenStateSec: Int = 30,
        /** 滑动窗口大小（调用次数）。 */
        val slidingWindowSize: Int = 20,
        /** 计算失败率所需的最少调用数。 */
        val minimumNumberOfCalls: Int = 10,
    )

    data class RetryConfig(
        val maxAttempts: Int = 2,
        val waitDurationMs: Long = 200,
    )
}
